lib/improver/blending/weights.py

```diff
@@ -244,6 +248,538 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+            }
+
+        """
+        self.config_dict = config_dict
+        self.weighting_coord_name = weighting_coord_name
+        self.config_coord_name = config_coord_name
+        self.weights_coord_name = weights_coord_name
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        msg = ("<ChooseWeightsLinearFromDict config_dict = {}, "
+               "weighting_coord_name = {}, config_coord_name = {}, "
+               "weights_coord_name = {}>".format(
+                   str(self.config_dict), self.weighting_coord_name,
+                   self.config_coord_name, self.weights_coord_name))
+        return msg
+
+    def _check_config_dict(self):
+        """Check whether the items within the configuration dictionary
+        are of matching lengths.
+
+        Raises:
+            ValueError: If items within the configuration dictionary are
+                not of matching lengths.
+        """
+        # Check all keys
+        for key in self.config_dict.keys():
+            weighting_len = (
+                len(self.config_dict[key][self.weighting_coord_name]))
+            weights_len = (
+                len(self.config_dict[key][self.weights_coord_name]))
+            if weighting_len != weights_len:
+                msg = ("{} is {}, {} is {}."
+                       "These items in the configuration dictionary "
+                       "have different lengths i.e. {} != {}".format(
+                           self.weighting_coord_name,
+                           self.config_dict[key][self.weighting_coord_name],
+                           self.weights_coord_name,
+                           self.config_dict[key][self.weights_coord_name],
+                           weighting_len, weights_len))
+                raise ValueError(msg)
+
+    def _arrange_interpolation_inputs(self, cube):
+        """Organise the inputs required for the linear interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the diagnostic that will be interpolated.
+
+        Returns:
+            (tuple): tuple containing
+
+                **source_points** (np.ndarray):
+                    Points within the configuration dictionary that will
+                    be used as the input to the interpolation.
+
+                **target_points** (np.ndarray):
+                    Points within the cube that will be the target points
+                    for the interpolation.
+
+                **associated_data** (np.ndarray):
+                    Weights from the configuration dictionary that will be
+                    used as the input to the interpolation.
+
+                **fill_value** (tuple):
+                    Values that be used if extrapolation is required. The
+                    fill values will be used as constants that are extrapolated
+                    if the target_points are outside the source_points
+                    provided.
```

Comment:
Add something like "These are the first and last weight values"

Reply:
Added.